truckersmp-cli

```diff
@@ -1,512 +1,547 @@
-            "STEAM_COMPAT_CLIENT_INSTALL_PATH=$XDG_DATA_HOME/Steam" \
-            "/usr/bin/env python3 $protondir/proton" \
-            "run" \
-            "$scriptdir/truckersmp-cli.exe $gamedir $moddir"
-
-    SteamGameId="$steamid" \
-    SteamAppId="$steamid" \
-    STEAM_COMPAT_DATA_PATH="$prefixdir" \
-    STEAM_COMPAT_CLIENT_INSTALL_PATH="$XDG_DATA_HOME/Steam" \
-    /usr/bin/env python3    "$protondir/proton" \
-                            run \
-                            "$scriptdir/truckersmp-cli.exe" \
-                            "$gamedir" \
-                            "$moddir"
-}
-
-start_with_wine() {
-cat <<WINE_USAGE
-
+            printerr("waiting 30 seconds for steam to start up.")
+            time.sleep(30)
+            printdbg("Steam should be up now")
+
+    if os.path.isdir(args.prefixdir):
+        printdbg("Creating directory {}".format(args.prefixdir))
+        os.makedirs(args.prefixdir, exist_ok = True)
+
+    env = {
+        "SteamGameId" : args.steamid,
+        "SteamAppId" : args.steamid,
+        "STEAM_COMPAT_DATA_PATH" : args.prefixdir,
+        "STEAM_COMPAT_CLIENT_INSTALL_PATH" : steamdir,
+    }
+    for key, value in env.items():
+        os.environ[key] = value
+    proton = os.path.join(args.protondir, "proton")
+    tmpcliexe = os.path.join(Dir.scriptdir, "truckersmp-cli.exe")
+    printdbg("""Startup command:
+  SteamGameId={}
+  SteamAppId={}
+  STEAM_COMPAT_DATA_PATH={}
+  STEAM_COMPAT_CLIENT_INSTALL_PATH={}
+  {} {}
+  run
+  {} {} {}""".format(env["SteamGameId"], env["SteamAppId"],
+      env["STEAM_COMPAT_DATA_PATH"], env["STEAM_COMPAT_CLIENT_INSTALL_PATH"],
+      sys.executable, proton, tmpcliexe, args.gamedir, args.moddir))
+    subprocess.call(
+      [sys.executable, proton, "run", tmpcliexe, args.gamedir, args.moddir])
+
+def start_with_wine():
+    print("""
     ###################################################################
     #                                                                 #
     #  Please check wine steam is running or the launcher won't work  #
     #                                                                 #
     ###################################################################
 
-    Press enter if you are good to go:
-WINE_USAGE
-    read -r _
-
-    print "Startup command:"
-    print   "WINEDEBUG=-all" \
-            "WINEARCH=win64" \
-            "WINEPREFIX=$prefixdir" \
-            "wine $scriptdir/truckersmp-cli.exe $gamedir $moddir"
-
-    # truckersmp-cli.exe GAMEDIR MODDIR
-    WINEDEBUG=-all \
-    WINEARCH=win64 \
-    WINEPREFIX="$prefixdir" \
-    wine    "$scriptdir/truckersmp-cli.exe" \
-            "$gamedir" \
-            "$moddir"
-}
+    Press enter if you are good to go: """, end = "")
+    sys.stdin.readline()
+
+    env = {
+        "WINEDEBUG" : "-all",
+        "WINEARCH" : "win64",
+        "WINEPREFIX" : args.prefixdir,
+    }
+    for key, value in env.items():
+        os.environ[key] = value
```

Comment:
Looks like you're overwriting the users environment here. Maybe it's a good idea to temporary save them and reset them afterwards to the old values.